### PR TITLE
Feature/app-secure view

### PR DESCRIPTION
**Pull request by tleekkul**
Added `app secure` view that works like a secure view but instead of validate with login JWT secret, it validates with application Token instead.

### CODE REVIEW

File: Configuration/odin-app.json [hunk 1]
```diff
@@ -17,6 +17,22 @@
                     }
                 }
             }
+        },
+        "views/odin/app/secure": {
```
Comment (KayEss) [inline Configuration/odin-app.json]: I don't think we need this in this config file. The view doesn't really do anything does it? If you need this for a test it should go in a test view JSON config file.

File: Cpp/odin-views/app_secure.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            const fostlib::host &host
+        ) const {
+            const auto paths = fostlib::split(path, "/");
+            if ( paths.size() != 1)
+                throw fostlib::exceptions::not_implemented(__func__,
+                    "Must pass app_id in the URL");
+            fostlib::pg::connection cnx{fostgres::connection(config, req)};
+            const auto app_id = paths[0];
+            fostlib::json app = odin::app::get_detail(cnx, app_id);
+            cnx.commit();
```
Comment (KayEss) [inline Cpp/odin-views/app_secure.cpp]: If you've not made any changes then there's no reason to do a commit here.

File: Schema/app/002-initial.blue.sql [hunk 3]
```diff
@@ -4,20 +4,25 @@ INSERT INTO odin.migration VALUES('app', '002-initial.blue.sql');
...
+    app_id TEXT NOT NULL,
     app_name TEXT NOT NULL,
+    token TEXT NOT NULL,
     access_policy TEXT NOT NULL,
     data_sharing_policy TEXT NOT NULL,
     redirect_url TEXT NOT NULL,
     changed TIMESTAMP WITH TIME ZONE DEFAULT NOW(),
     created TIMESTAMP WITH TIME ZONE DEFAULT NOW(),
-    PRIMARY KEY (app_id)
+    PRIMARY KEY (app_id),
```
Comment (KayEss) [inline Schema/app/002-initial.blue.sql]: Should put these with the fields that they are talking about. This line and the ones below for the foreign key should go here under the `app_id` column.

Comment (tleekkul) [reply Schema/app/002-initial.blue.sql]: Disagree, I think this style is much easier for reading and reasoning.

Comment (KayEss) [reply Schema/app/002-initial.blue.sql]: You're wrong :-) But let's discuss offline. Short argument is that nowhere else in the code would you want to put closely related items far apart from each other.

File: Schema/app/002-initial.blue.sql [hunk 4]
```diff
@@ -104,33 +110,39 @@ EXECUTE PROCEDURE odin.app_owner_ledger_insert();
...
+
+    app_id TEXT NOT NULL,
     FOREIGN KEY (app_id)
         REFERENCES odin.app (app_id) MATCH SIMPLE
-        ON UPDATE NO ACTION ON DELETE NO ACTION DEFERRABLE
+        ON UPDATE NO ACTION ON DELETE NO ACTION DEFERRABLE,
+
+    PRIMARY KEY (reference, identity_id, app_id),
+
+    state TEXT NOT NULL DEFAULT 'ACTIVE',
```
Comment (KayEss) [inline Schema/app/002-initial.blue.sql]: Should there not be a constraint over these `state` columns with the allowed state values?

Comment (tleekkul) [reply Schema/app/002-initial.blue.sql]: I think for now it's OK because we are not sure about the choices of `state` yet.

**Pull request closed without merging**